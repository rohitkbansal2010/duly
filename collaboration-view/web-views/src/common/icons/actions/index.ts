import addIcon from './add.svg';
import AfterVisitLinkIcon from './after-visit.svg';
import attachIcon from './attach.svg';
import browseIcon from './browse.svg';
import closeIcon from './close.svg';
import cropIcon from './crop.svg';
import downloadIcon from './download.svg';
import editIcon from './edit.svg';
import expandSolidIcon from './expand-solid.svg';
import expandIcon from './expand.svg';
import ExportPdf from './export-pdf.svg';
import favoriteIcon from './favorite.svg';
import filterOutlinedIcon from './filter-outlined.svg';
import filterSolidIcon from './filter-solid.svg';
import FontIconBlue from './font-icon-blue.svg';
import FontIconSkyblue from './font-icon-skyblue.svg';
import lineToolIcon from './line-tool.svg';
import Line from './Line.svg';
import messageIcon from './message.svg';
import moreIcon from './more.svg';
import noteIcon from './note.svg';
import printIcon from './print.svg';
import reflectIcon from './reflect.svg';
import reviewVitalsIcon from './review-vitals.svg';
import searchIcon from './search.svg';
import sendIcon from './send.svg';
import shapeToolIcon from './shape-tool.svg';
import textToolIcon from './text-tool.svg';
import updateIcon from './update.svg';
import uploadIcon from './upload.svg';
import viewPdfIcon from './view-pdf.svg';
import voiceRecorderIcon from './voice-recorder.svg';

export {
  reviewVitalsIcon,
  viewPdfIcon,
  voiceRecorderIcon,
  textToolIcon,
  updateIcon,
  uploadIcon,
  shapeToolIcon,
  searchIcon,
  sendIcon,
  printIcon,
  reflectIcon,
  addIcon,
  noteIcon,
  lineToolIcon,
  messageIcon,
  moreIcon,
  filterSolidIcon,
  attachIcon,
  filterOutlinedIcon,
  browseIcon,
  closeIcon,
  cropIcon,
  downloadIcon,
  favoriteIcon,
  editIcon,
  expandIcon,
  expandSolidIcon,
  FontIconSkyblue,
  FontIconBlue,
  AfterVisitLinkIcon,
  ExportPdf,
  Line
};
