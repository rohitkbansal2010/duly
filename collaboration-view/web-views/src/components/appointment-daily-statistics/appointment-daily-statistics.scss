@import "~@styles/colors.scss";
@import "~@styles/media-mixins.scss";
@import "~@styles/spacing-constants.scss";
@import "~@styles/main.scss";
@import "~@styles/typography.scss";
@import "~@styles/devices.scss";

.topSection {
  display: flex;
  flex-direction: column;
  gap: 6rem;
  padding-bottom: 1.25rem;
  border-bottom: 0.0625rem solid rgba($medium-gray, 0.5);
}

.dateRow {
  line-height: 4rem;

  &Day {
    color: $dupage-violet;
    font-size: 3.1rem;
  }

  &Month {
    @include nexa-heavy-highlight;
    color: $dupage-magenta;
    font-size: 4.5rem;
  }

  &Year {
    color: $medium-gray;
    font-size: 2.15rem;
    line-height: 3.5rem;
  }
}

.addressRow {
  color: $medium-gray;
  line-height: 1.8rem;
  cursor: pointer;

  &TopPart {
    position: relative;
    padding-left: 3.2rem !important;
    font-size: 1.5rem;
    @include nexa-heavy-highlight;

    & img {
      position: absolute;
      left: 0.5rem;
      width: 2rem;
    }

    &SpacingSpan {
      padding-right: 0.2rem;
    }
  }

  &BottomPart {
    padding-left: 3.2rem !important;
    font-size: 1.15rem;
  }
}

.bottomSection {
  padding-top: 0.75rem;
}
