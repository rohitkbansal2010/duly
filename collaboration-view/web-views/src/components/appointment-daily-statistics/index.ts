export { AppointmentDailyStatistics } from './appointment-daily-statistics';
