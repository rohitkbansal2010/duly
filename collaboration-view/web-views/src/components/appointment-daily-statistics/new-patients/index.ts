export { NewPatients } from './new-patients';
