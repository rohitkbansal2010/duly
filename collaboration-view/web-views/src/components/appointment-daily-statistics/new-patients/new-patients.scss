@import "~@styles/colors.scss";
@import "~@styles/media-mixins.scss";
@import "~@styles/spacing-constants.scss";
@import "~@styles/main.scss";
@import "~@styles/typography.scss";
@import "~@styles/devices.scss";

.newPatientsRow {
  font-size: 1.5rem;
  line-height: 1.8rem;
  white-space: nowrap;
  @include nexa-heavy-highlight;

  &Container {
    position: relative;
    padding-left: 3.2rem !important;
  }

  & img {
    position: absolute;
    bottom: 0;
    left: 0.5rem;
    width: 2.6rem;
    padding: 0 0.3rem;
  }

  &Count {
    color: $dupage-dark-blue;
  }

  &Label {
    margin-left: 0.4rem;
    color: $medium-gray;
  }
}
