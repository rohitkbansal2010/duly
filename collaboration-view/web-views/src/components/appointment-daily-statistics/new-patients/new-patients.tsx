import { Row } from 'react-bootstrap';

import { newPatientsMagenta } from '@icons';
import { DailyStatistics } from '@types';

import styles from './new-patients.scss';

type NewPatientsTypeProps = Pick<DailyStatistics, 'newPatients'>;

export const NewPatients = ({ newPatients }: NewPatientsTypeProps) => {
  if (!newPatients) return null;

  const newPatientsLabel = newPatients > 1 ? 'New Patients' : 'New Patient';

  return (
    <Row className={styles.newPatientsRow}>
      <span className={styles.newPatientsRowContainer}>
        <img src={newPatientsMagenta} alt="new patients icon" />
        <span className={styles.newPatientsRowCount}>{newPatients}</span>
        <span className={styles.newPatientsRowLabel}>{newPatientsLabel}</span>
      </span>
    </Row>
  );
};
