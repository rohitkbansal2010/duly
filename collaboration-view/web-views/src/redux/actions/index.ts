export * from './ui';
export * from './locale';
export * from './types';
export * from './user';
export * from './app-state';
export * from './todays-appointments';
export * from './current-appointment';
export * from './overview-widgets';
export * from './site';
