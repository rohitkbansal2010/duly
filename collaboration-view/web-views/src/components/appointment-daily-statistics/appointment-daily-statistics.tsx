import { useCallback } from 'react';
import { Row, Col } from 'react-bootstrap';
import { useDispatch, useSelector } from 'react-redux';

import { SitesContainer } from '@components/sites-container';
import { MONTH_NAMES_SHORT, WEEKDAYS_NAMES } from '@constants';
import { mapPinMagenta } from '@icons';
import { showLocationPickerModal } from '@redux/actions';
import { RootState } from '@redux/reducers';
import { formatAddressBottomLine } from '@utils';

import styles from './appointment-daily-statistics.scss';
import { NewPatients } from './new-patients';

export const AppointmentDailyStatistics = () => {
  const dispatch = useDispatch();
  const today = new Date();

  const dailyStatistics = useSelector(({ TODAYS_APPOINTMENTS }: RootState) =>
    TODAYS_APPOINTMENTS.dailyStatistics);
  const currentSite = useSelector(({ SITE }: RootState) =>
    SITE.currentSite);

  const addressBottomLine = currentSite?.address ? formatAddressBottomLine(currentSite.address) : '';

  const handleClick = useCallback(() => {
    dispatch(showLocationPickerModal());
  }, [ dispatch ]);

  const newPatients = dailyStatistics && dailyStatistics.newPatients ?
    <NewPatients newPatients={dailyStatistics.newPatients} /> : null;

  return (
    <Col>
      <div className={styles.topSection}>
        <Row className={styles.dateRow}>
          <span className={styles.dateRowDay} data-testid="dateRowDay">{WEEKDAYS_NAMES[today.getDay()]},</span>
          <span className={styles.dateRowMonth} data-testid="dateRowMonth">
            {today.getDate()} {MONTH_NAMES_SHORT[today.getMonth()]}
          </span>
          <span className={styles.dateRowYear} data-testid="dateRowYear">{today.getFullYear()}</span>
        </Row>
        <Row onClick={handleClick} className={styles.addressRow} data-testid="addressRow">
          {currentSite && (
            <SitesContainer>
              <span className={styles.addressRowTopPart} data-testid="addressRowTopPart">
                <img src={mapPinMagenta} alt="map pin magenta" />
                <span className={styles.addressRowTopPartSpacingSpan} data-testid="addressRowTopPartSpacingSpan">
                  {currentSite?.address?.line}
                </span>
              </span>
              <span className={styles.addressRowBottomPart} data-testid="addressRowBottomPart">
                {addressBottomLine}
              </span>
            </SitesContainer>
          )}
        </Row>
      </div>
      <div className={styles.bottomSection}>
        {newPatients}
      </div>
    </Col>
  );
};
