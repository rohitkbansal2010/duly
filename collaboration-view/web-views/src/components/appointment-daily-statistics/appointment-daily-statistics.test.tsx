import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';
import { useSelector } from 'react-redux';

import { MONTH_NAMES_SHORT, WEEKDAYS_NAMES } from '@constants';
import { mapPinMagenta } from '@icons';
import { showLocationPickerModal } from '@redux/actions';
import { Site } from '@types';

import { AppointmentDailyStatistics } from './appointment-daily-statistics';
import { NewPatients } from './new-patients';

const mockDispatch = jest.fn();
jest.mock('react-redux', () =>
  ({
    ...jest.requireActual('react-redux'),
    useDispatch: () =>
      mockDispatch,
    useSelector: jest.fn(),
  }));

const mockSite: Site = {
  id: '111',
  address: {
    city: 'Lunacity',
    line: 'FirstLine',
    postalCode: 'Lu001',
    state: 'Lunastate',
  },
};

describe('AppointmentDailyStatistics', () => {
  let dateSpy: jest.SpyInstance;
  beforeAll(() => {
    const mockDate = new Date('2022-11-01');
    dateSpy = jest.spyOn(global, 'Date');
    dateSpy.mockImplementation(() =>
      mockDate);
  });
  afterAll(() => {
    dateSpy.mockRestore();
  });

  let wrapper: ShallowWrapper;
  beforeEach(() => {
    (useSelector as jest.Mock).mockImplementation((selector: (state: unknown) => unknown) =>
      selector({
        TODAYS_APPOINTMENTS: { dailyStatistics: { newPatients: 3 } },
        SITE:{ currentSite: mockSite },
      }));
    wrapper = shallow(<AppointmentDailyStatistics />);
  });

  it('should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should pass correct props in NewPatients component', () => {
    const newPatientsComponent = wrapper.find(NewPatients);

    expect(newPatientsComponent.props()).toEqual({ newPatients: 3 });
  });

  it('should not render NewPatients component if newPatients = 0', () => {
    (useSelector as jest.Mock).mockImplementation((selector: (state: unknown) => unknown) =>
      selector({
        TODAYS_APPOINTMENTS: { dailyStatistics: { newPatients: 0 } },
        SITE:{ currentSite: mockSite },
      }));
    wrapper = shallow(<AppointmentDailyStatistics />);

    expect(wrapper.find(NewPatients).length).toBe(0);
  });

  it('should render correct weekday in dateRowDay element', () => {
    const weekday = WEEKDAYS_NAMES[new Date().getDay()];
    const renderedWeekday = wrapper.find({ 'data-testid': 'dateRowDay' }).text()
      .slice(0, -1);

    expect(renderedWeekday).toEqual(weekday);
  });

  it('should render correct date in dateRowMonth element', () => {
    const date = new Date().getDate();
    const renderedDateMatchArray = wrapper.find({ 'data-testid': 'dateRowMonth' }).text()
      .match(/\d+/);
    const renderedDate = renderedDateMatchArray && renderedDateMatchArray[0];

    expect(renderedDate).toEqual(String(date));
  });

  it('should render correct month in dateRowMonth element', () => {
    const month = MONTH_NAMES_SHORT[new Date().getMonth()];
    const renderedMonth = wrapper.find({ 'data-testid': 'dateRowMonth' }).text()
      .replace(/[\d|\s]+/, '');

    expect(renderedMonth).toEqual(month);
  });

  it('should render correct year in dateRowYear element', () => {
    const year = new Date().getFullYear();
    const renderedYear = wrapper.find({ 'data-testid': 'dateRowYear' }).text();

    expect(renderedYear).toEqual(String(year));
  });

  it('should call showLocationPickerModal on addressRow element click', () => {
    const addressRowElement = wrapper.find({ 'data-testid': 'addressRow' });
    addressRowElement.simulate('click');

    expect(mockDispatch).toHaveBeenCalledWith(showLocationPickerModal());
  });

  it('should render image in addressRowTopPart element with src = mapPinMagenta', () => {
    const addressRowTopPartElement = wrapper.find({ 'data-testid': 'addressRowTopPart' });
    const image = addressRowTopPartElement.find('img');

    expect(image.prop('src')).toEqual(mapPinMagenta);
  });

  it('should render correct address line in addressRowTopPartSpacingSpan element', () => {
    const line = mockSite?.address?.line;
    const renderedLine = wrapper.find({ 'data-testid': 'addressRowTopPartSpacingSpan' }).text();

    expect(renderedLine).toEqual(line);
  });

  it('should render correct address city, state and postalCode in addressRowBottomPart element', () => {
    const renderedText = wrapper.find({ 'data-testid': 'addressRowBottomPart' }).text();

    expect(renderedText)
      .toEqual(`${mockSite?.address?.city}, ${mockSite?.address?.state} ${mockSite?.address?.postalCode}`);
  });
});
