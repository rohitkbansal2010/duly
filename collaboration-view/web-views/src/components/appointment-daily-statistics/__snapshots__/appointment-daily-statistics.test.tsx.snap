// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppointmentDailyStatistics should match snapshot 1`] = `
<Col>
  <div
    className="topSection"
  >
    <Row
      className="dateRow"
    >
      <span
        className="dateRowDay"
        data-testid="dateRowDay"
      >
        Tuesday
        ,
      </span>
      <span
        className="dateRowMonth"
        data-testid="dateRowMonth"
      >
        1
         
        Nov
      </span>
      <span
        className="dateRowYear"
        data-testid="dateRowYear"
      >
        2022
      </span>
    </Row>
    <Row
      className="addressRow"
      data-testid="addressRow"
      onClick={[Function]}
    >
      <SitesContainer>
        <span
          className="addressRowTopPart"
          data-testid="addressRowTopPart"
        >
          <img
            alt="map pin magenta"
            src="map-pin-magenta.svg"
          />
          <span
            className="addressRowTopPartSpacingSpan"
            data-testid="addressRowTopPartSpacingSpan"
          >
            FirstLine
          </span>
        </span>
        <span
          className="addressRowBottomPart"
          data-testid="addressRowBottomPart"
        >
          Lunacity, Lunastate Lu001
        </span>
      </SitesContainer>
    </Row>
  </div>
  <div
    className="bottomSection"
  >
    <NewPatients
      newPatients={3}
    />
  </div>
</Col>
`;
